src/aggregate/aggregate_request.c

```diff
@@ -473,34 +491,44 @@ static int parseGroupby(AREQ *req, ArgsCursor *ac, QueryError *status) {
   }
 
   // Number of fields.. now let's see the reducers
-
-  PLN_GroupStep *gstp = calloc(1, sizeof(*gstp));
-  gstp->properties = (const char **)groupArgs.objs;
-  gstp->nproperties = groupArgs.argc;
-  gstp->base.dtor = groupStepFree;
-  gstp->base.type = PLN_T_GROUP;
+  PLN_GroupStep *gstp = PLNGroupStep_New((const char **)groupArgs.objs, groupArgs.argc);
   AGPLN_AddStep(&req->ap, &gstp->base);
 
   while (AC_AdvanceIfMatch(ac, "REDUCE")) {
-    PLN_Reducer *cur;
     const char *name;
     if (AC_GetString(ac, &name, NULL, 0) != AC_OK) {
       QERR_MKBADARGS_AC(status, "REDUCE", rv);
       return REDISMODULE_ERR;
     }
-    cur = array_ensure_tail(&gstp->reducers, PLN_Reducer);
-
-    if (buildReducer(gstp, cur, ac, name, status) != REDISMODULE_OK) {
+    if (PLNGroupStep_AddReducer(gstp, name, ac, status) != REDISMODULE_OK) {
       goto error;
     }
   }
-  gstp->idx = req->serial++;
   return REDISMODULE_OK;
 
 error:
   return REDISMODULE_ERR;
 }
 
+static void freeFilterStep(PLN_BaseStep *bstp) {
+  PLN_MapFilterStep *fstp = (PLN_MapFilterStep *)bstp;
+  if (fstp->parsedExpr) {
+    ExprAST_Free(fstp->parsedExpr);
+  }
+  if (fstp->shouldFreeRaw) {
+    free((char *)fstp->rawExpr);
+  }
+  free(bstp);
+}
+
+PLN_MapFilterStep *PLNMapFilterStep_New(const char *expr, int mode) {
+  PLN_MapFilterStep *stp = calloc(1, sizeof(*stp));
```

Comment:
make sure to use rm_malloc from now on, we should not allocate memory behind the redis back ..